### ISSUE TITLE
gcc version detection seems to be broken

### ISSUE THREAD

**Issue by coiouhkc** [OpenDUNE/OpenDUNE#381]
See #290,

```
gcc -dumpversion
12
```

**Comment by coiouhkc** [comment]
Suggested fix/workaround: 
```
git diff --patch-with-stat
 config.lib | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/config.lib b/config.lib
index 09bd456e..4d376be4 100644
--- a/config.lib
+++ b/config.lib
@@ -728,11 +728,12 @@ make_compiler_cflags() {
                # rdynamic is used to get useful stack traces from crash reports.
                ldflags="$ldflags -rdynamic"
        else
-               # Enable some things only for certain GCC versions
+                # Enable some things only for certain GCC versions
+                cc_version_orig=`$1 -dumpversion`
                cc_version=`$1 -dumpversion | cut -c 1,3`
                if [ $(echo -n "$cc_version" | wc -c) -eq 1 ] ; then
                        # add a 0 if the version had only one digit
-                       cc_version="${cc_version}0"
+                       cc_version="${cc_version_orig}0"
                fi
 
                if [ $cc_version -lt 33 ]; then
```

**Comment by miniupnp** [comment]
```
nanard@openbox|13:24:29|~$ gcc -dumpversion | cut -c 1,3
33
nanard@openbox|13:24:39|~$ gcc -dumpversion | cut -d'.' -f1-2 | tr -d '.'
33
```

**Comment by miniupnp** [comment]
I believe I shoud do
```
cc_version=`$1 -dumpversion | cc_version=`$1 -dumpversion | cut -c 1,3`
if echo "$cc_version" | grep -q '\.' ; then
  cc_version="${cc_version}.0"
fi
cc_version=`echo "$cc_version" | tr -d '.'` 
```

**Comment by miniupnp** [comment]
it was already fixed with 411c3a9d2608767f6491d2f2da441bcb84fe64d4
see #356
